Odds-and-ends of Nickel formatting

Xophmeister opened the pull request:
* [x] Conditionals
* [x] Annotations (again)
  * [x] Not reflowed on to a new line if:
    * [x] There is no equal sign, or
    * [x] The equal sign is "blocked" by some other node (e.g., a comment)
  * [x] Idempotency of multi-line nested type annotations
* [x] Multi-line sequences of infix operators
* [x] Formatting for `let...in...` expressions
* [x] Consistent formatting of let expressions and record fields
* [x] No spacing after prefix operators (`-` and `!`)
* [x] Generalise `(match)` and `(destruct)` statements to use record formatting rules

**Note** There is an idempotency issue with `let` expressions. However, the case in which it occurs is quite specific; it is not, for example, manifested by the Nickel standard library. It is documented in the query file and I will add an issue, but it shouldn't be a blocker for this PR.

torhovland commented on languages/nickel.scm:
```diff
@@ -86,83 +87,153 @@
+)
+
+(let_in_block
+  (#scope_id! "let_binding_rhs")
```
It seems that a scope isn't necessary here.

torhovland commented on languages/nickel.scm:
```diff
@@ -86,83 +87,153 @@
+)
+
+(let_expr
+  (#scope_id! "let_result")
```
It seems that a scope isn't necessary here either.

torhovland commented on languages/nickel.scm:
```diff
@@ -86,83 +87,153 @@
+  (infix_expr) @begin_scope
+) @end_scope
+
+(infix_expr
```
This pattern doesn't seem to make any difference in the sample file.

torhovland commented on languages/nickel.scm:
```diff
@@ -86,83 +87,153 @@
-;   }
+; context.
 (
   (#scope_id! "signature")
```
It seems that a scope isn't necessary here either.

torhovland commented on languages/nickel.scm:
```diff
@@ -187,11 +258,36 @@
+;; Conditionals
+
+; Flow multi-line match cases into an indented block after the =>
+(match_case
```
This pattern doesn't seem to make any difference in the sample file.

torhovland reviewed (commented):
Looks like we either have unnecessary patterns or we are missing some sample code.

Xophmeister commented:
I've updated the tests with the concatenated Nickel stdlib. This now hits every query in the Nickel formatting definitions.

I've also changed the `let...in...` indentation, mentioned by the Nickel team. I've yet to fix [this](https://github.com/tweag/nickel/pull/1198#discussion_r1147862341), because I'm still trying to find a minimal reproducing example. (When I do, I'll add it to the tests :innocent:)

aspiwack commented:
As always, we should merge this earlier rather than later as it's clear improvement. And when it's done, we can document what we want to improve further.
